sdk/core/azure-core/inc/azure/core/http/winhttp/win_http_client.hpp

```diff
@@ -6,29 +6,103 @@
  */
 
 #pragma once
-#ifdef BUILD_TRANSPORT_WINHTTP_ADAPTER
+//#ifdef BUILD_TRANSPORT_WINHTTP_ADAPTER
 
 #include "azure/core/http/http.hpp"
 #include "azure/core/http/policy.hpp"
 
 #include <type_traits>
 #include <vector>
+#include <winhttp.h>
 
 namespace Azure { namespace Core { namespace Http {
 
+  namespace Details {
+
+    constexpr static int64_t DefaultUploadChunkSize = 1024 * 64;
+    constexpr static int64_t MaximumUploadChunkSize = 1024 * 1024 * 1024;
+
+    class WinHttpStream : public BodyStream {
+    private:
+      HINTERNET m_sessionHandle;
+      HINTERNET m_connectionHandle;
+      HINTERNET m_requestHandle;
+      bool m_isEOF;
+
+      /**
+       * @brief This is a copy of the value of an HTTP response header `content-length`. The value
+       * is received as string and parsed to size_t. This field avoid parsing the string header
+       * every time from HTTP RawResponse.
+       *
+       * @remark This value is also used to avoid trying to read more data from network than what we
+       * are expecting to.
+       *
+       * @remark A value of -1 means the transfer encoding was chunked.
+       *
+       */
+      int64_t m_contentLength;
+
+      int64_t m_streamTotalRead;
```

Comment:
totalread fits better with the rest of the local variable names and how this integer is being used as an accumulator since I am not passing this to any streaming API directly. Leaving as is.